kong/router/fields.lua

```diff
@@ -0,0 +1,346 @@
+local buffer = require("string.buffer")
+local context = require("resty.router.context")
+
+
+local type = type
+local ipairs = ipairs
+local fmt = string.format
+local tb_sort = table.sort
+local tb_concat = table.concat
+local replace_dashes_lower = require("kong.tools.string").replace_dashes_lower
+
+
+local var           = ngx.var
+local get_method    = ngx.req.get_method
+local get_headers   = ngx.req.get_headers
+local get_uri_args  = ngx.req.get_uri_args
+local server_name   = require("ngx.ssl").server_name
+
+
+local PREFIX_LEN = 13 -- #"http.headers."
+local HTTP_HEADERS_PREFIX = "http.headers."
+local HTTP_QUERIES_PREFIX = "http.queries."
+
+
+local FIELDS_FUNCS = {
+    -- http.*
+
+    ["http.method"] =
+    function(params)
+      if not params.method then
```

Comment:
It seems that `params.method` will never be `nil`?

Reply:
This is for unit-testing.

Reply:
Is this still needed after the change?

Reply:
Yes, we will acquire `params.method` in `get_cache_key()`, so we should check if it is `nil`.